mempool/src/shared_mempool/network.rs

```diff
@@ -79,7 +79,7 @@ pub enum BroadcastError {
 #[derive(Clone, Debug)]
 pub(crate) struct MempoolNetworkInterface<NetworkClient> {
     network_client: NetworkClient,
-    sync_states: Arc<LockingHashMap<PeerNetworkId, PeerSyncState>>,
```

Comment:
Do you have context on why this was added in the first place? Looking at the code, it doesn't look like adding anything other than introducing another layer of indirection.

Reply:
IIRC, this was added a while back (1.5 years??), as part of a refactor that was *going* to take place, but it ended up being deprioritized. 😄